### PR TITLE
adios2: add latest release

### PR DESCRIPTION
**Pull request by vicentebolea**
:tada: :tada: :tada: 

Finally! ADIOS v2.9.0 final release is out.

This bring exiting new features and bugfixes. Find the release at https://github.com/ornladios/ADIOS2/releases/tag/v2.9.0

Let me know if we need to change anything else in the package

ViB

:tada: :tada: :tada:

### CODE REVIEW

**Review by tldahlgren — DISMISSED** [review]
Confirmed the sha256.

**Comment by vicentebolea** [comment]
@spackbot fix style

**Comment by MichaelLaufer** [comment]
@vicentebolea, thanks for this! I believe that the package is currently missing some new optional dependencies: `c-blosc2`, and `ucx@1.9.0:`, and `kokkos`.

**Comment by vicentebolea** [comment]
> `c-blosc2`, and `ucx@1.9.0:`, and `kokkos`.

Lets do that in another PR. Here we need the fix the build issues first.

**Comment by vicentebolea** [comment]
@spackbot fix style

**Comment by vicentebolea** [comment]
@spackbot fix style

**Comment by vicentebolea** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/sensei/package.py [hunk 1]
```diff
@@ -98,6 +98,8 @@ class Sensei(CMakePackage):
     conflicts("+ascent", when="+libsim")
 
     # Patches
+    # https://github.com/SENSEI-insitu/SENSEI/pull/114
+    patch("adios2-remove-deprecated-functions.patch", when="@4 ^adios2@2.9:")
```
Comment (kwryankrattiger) [inline var/spack/repos/builtin/packages/sensei/package.py]: Can this patch be for `@4:4.1` since I think that PR is pretty close to going in.

Comment (vicentebolea) [reply var/spack/repos/builtin/packages/sensei/package.py]: Thanks for pointing this out. Edited in the last push

**Comment by kwryankrattiger** [comment]
Patch failed to apply to SENSEI
https://gitlab.spack.io/spack/spack/-/jobs/6631818

**Comment by vicentebolea** [comment]
@kwryankrattiger, unfortunately there are conflicts on this patch when applying to sensei 4|4.1, I went ahead and resolved these conflicts and created a new patch that only works with 4/4.1 which should work. On the other hand, there is no way to submit this upstream since it appears that sensei does not maintain releases.

**Comment by vicentebolea** [comment]
@kwryankrattiger Ready for re-review

**Review by kwryankrattiger — APPROVED** [review]
Thanks @vicentebolea

**Pull request merged**